invenio_vocabularies/contrib/subjects/datastreams.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 University of Münster.
+#
+# Invenio-Vocabularies is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Names datastreams, transformers, writers and readers."""
+
+from invenio_access.permissions import system_identity
+from invenio_i18n import lazy_gettext as _
+
+from ...datastreams.writers import ServiceWriter
+from .mesh.datastreams import VOCABULARIES_DATASTREAM_READERS as mesh_readers
+from .mesh.datastreams import VOCABULARIES_DATASTREAM_TRANSFORMERS as mesh_transformers
+from .mesh.datastreams import VOCABULARIES_DATASTREAM_WRITERS as mesh_writers
+
+
+class SubjectsServiceWriter(ServiceWriter):
+    """Subjects Service Writer."""
+
+    def __init__(self, *args, **kwargs):
+        """Constructor."""
+        service_or_name = kwargs.pop("service_or_name", "subjects")
+        super().__init__(service_or_name=service_or_name, *args, **kwargs)
+
+    def _entry_id(self, entry):
+        """Get the id from an entry."""
+        return entry["id"]
```

Comment:
comment: if this subclass only exists because of this part, we should have a look at how to make this generic/easier to configure, since just subclassing for this reasons is very inefficient

Reply:
It confused me as well, but I guess updating it might need a major? (but it's a "private" function) so maybe not really?

I think all vocab writers can use this change. Additionally, this part is also funny to me: https://github.com/inveniosoftware/invenio-vocabularies/blob/570f268414b1aece6dc60c1b274e4de29d873f0c/invenio_vocabularies/contrib/affiliations/datastreams.py#L22-L25

to register just a specific vocab writer, we do this, I would just prefer using the service writer with the name/just fix the name and don't care about the arg.